frontend/assets/user_defined_in_basic.js

```diff
@@ -0,0 +1,42 @@
+function UserDefinedInBasic() {
+}
+
+UserDefinedInBasic.prototype.init = function(record_type, fields, read_only_view) {
```

Comment:
Passing stuff in here from the controller.  Hmm... actually I don't even use record_type, so maybe we could drop it.  Maybe I won't just now because I'm not sure if it'll invalidate my comments on this PR :)

The `fields` array is now a list of the i18N labels of the fields we want to pull up.  That's gross, but as you noted there's nothing better on the read-only view we can match on.  In the spectrum of evil I felt like doing things this way was less evil than copy/pasting bits of the form helper.

And read_only_view is just a boolean that we use to work out how to manipulate the DOM.  I found on the accessions read-only page that I needed to rewrite some Bootstrap classes to make things display consistently.  Bleh again.